compiler/block.py

```diff
@@ -440,16 +440,29 @@ def __init__(self, node):
     BlockVisitor.__init__(self)
     self.is_generator = False
     node_args = node.args
-    args = [a.arg for a in node_args.args]
+    args = []
+    for arg in node_args.args:
+      if isinstance(arg, ast.Tuple):
+        args.append([a.arg for a in arg.elts])
+      else:
+        args.append(arg.arg)
     if node_args.vararg:
       args.append(node_args.vararg.arg)
     if node_args.kwarg:
       args.append(node_args.kwarg.arg)
     for i, name in enumerate(args):
-      if name in self.vars:
-        msg = "duplicate argument '{}' in function definition".format(name)
-        raise util.ParseError(node, msg)
-      self.vars[name] = Var(name, Var.TYPE_PARAM, arg_index=i)
+      if isinstance(name, list):
+        self.add_vars(i, 'τ'.join(name), node)
```

Comment:
You can allocate a Go variable to hold the temporary value. Look for uses of Block.alloc_temp() in the code. The way this works for assignments is very similar to what I imagine you will need to do. It's a little indirect, but here's the relevant alloc for normal assignments: https://github.com/google/grumpy/blob/master/compiler/stmt.py#L640

I think the code for tuple argument unpacking will be very similar to that for tuple assignments. Hopefully you can reuse _build_assign_target(), _tie_target(), etc.